translator/trace/zipkin/zipkinv2_to_traces.go

```diff
@@ -428,3 +429,20 @@ func extractInstrumentationLibrary(zspan *zipkinmodel.SpanModel) string {
 	}
 	return zspan.Tags[conventions.InstrumentationLibraryName]
 }
+
+func setTimestampsV2(zspan *zipkinmodel.SpanModel, dest pdata.Span, destAttrs pdata.AttributeMap) {
+	// zipkin allows timestamp to be unset, but opentelemetry-collector expects it to have a value.
+	// unset gets converted to zero on json deserialization because time.Time (the type of
+	// Timestamp field) cannot be nil.  If timestamp is zero, the conversion from this
+	// internal format back to zipkin format in zipkin exporter fails
+	if zspan.Timestamp.IsZero() {
+		now := pdata.TimestampFromTime(time.Now())
```

Comment:
is it ok to treat this as now? maybe convert both to 0 (unixtime 0 not time 0) which is OTLP default? Maybe on the traces_to_zipkinv2 we also check and if they are 0 (unixtime 0 not time 0) we don't set the time?

This way we also handle missing case for OTLP?

Reply:
@bogdandrutu Is this the only blocker to get this merged?
Would it not be expected that the time the span is received is relatively close to the time it was originally created and thus would serve as a better timestamp for downstream searching than unixtime 0?

Reply:
I think @bogdandrutu had a good point, and his suggested change would result in the outgoing zipkinv2 span looking the same as the incoming zipkinv2 span.  I'm less certain what would happen if the output was in a different format. You both likely know better than I.

I just finished an attempt at this which I'll push up... apologies it took so long.